.evergreen/config.yml

```diff
@@ -0,0 +1,176 @@
+            export ATLAS_API_PASSWORD="${atlas_secret}"
+          EOT
+          # Don't print this as it contains credentials.
+    - command: expansions.update
+      params:
+        file: astrolabe-expansion.yml
+    # Use script to install astrolabe.
+    - command: shell.exec
+      params:
+        working_dir: astrolabe-src
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          sh .evergreen/install-astrolabe.sh
+
+  "install driver":
+    # Clone driver source code.
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          git clone --recursive --branch "${DRIVER_REVISION}" "${DRIVER_REPOSITORY}" driver-src
+    # Make the driver's *.sh files executable.
+    - command: shell.exec
+      params:
+        working_dir: driver-src
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          for i in $(find .evergreen/atlas-testing -name \*.sh); do
+            chmod +x $i
+          done
+    # Setup driver-specific environment expansions.
+    - command: subprocess.exec
+      params:
+        working_dir: driver-src
+        binary: .evergreen/atlas-testing/write-driver-expansion.sh
+        add_expansions_to_env: true
+    - command: shell.exec
+      params:
+        script: |
+          cat driver-src/driver-expansion.yml
+    - command: expansions.update
+      params:
+        file: driver-src/driver-expansion.yml
+    # Run driver-supplied script to install the driver.
+    - command: shell.exec
+      params:
+        working_dir: driver-src
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          ${PREPARE_SHELL_DRIVER}
+          sh .evergreen/atlas-testing/install-driver.sh
+
+  "run tests":
+    # Run the Atlas Planned Maintenance Tests.
+    - command: shell.exec
+      type: test
+      params:
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          ${PREPARE_SHELL_DRIVER}
+          "$ASTROLABE_BINARY" spec-tests run astrolabe-src/tests -e driver-src/.evergreen/atlas-testing/workload-executor.sh
+
+  "upload test results":
+    # Upload the xunit-format test results.
+    - command: attach.xunit_results
+      params:
+        file: "xunit-output/*.xml"
+
+# Functions to run before the test.
+pre:
+  - func: "install astrolabe"
+  - func: "install driver"
+
+# Functions to run after the test.
+post:
+  - func: "upload test results"
+
+tasks:
+    # Generic task to be used by all drivers.
+    - name: "run-tests"
+      commands:
+        - func: "run tests"
+
+axes:
+  # The 'driver' axis specifies the driver to be tested (including driver version).
+  # Drivers MUST specify the DRIVER_REPOSITORY and DRIVER_REVISION variables.
+  - id: driver
+    display_name: driver
+    values:
+      - id: pymongo-master
+        display_name: "PyMongo (master)"
+        variables:
+          DRIVER_REPOSITORY: "https://github.com/prashantmital/mongo-python-driver.git"
+          DRIVER_REVISION: "PYTHON-1743/test-atlas-planned-maintenance"
```

Comment:
I thought the plan was to add the driver specific test scripts to this repo. Why did we decide to add the scripts to the driver repo instead? This makes it more difficult to follow the flow because it jumps back and forth between multiple files in two repos, eg PYTHON_BINARY is defined in 'runtime' axis and used in `mongo-python-driver/.evergreen/atlas-testing/install-driver.sh`.

Reply:
Yes it definitely makes it harder to follow the flow, but this was done to minimize code churn in the driver-atlas-testing repo when more drivers start integrating this testing into their CI workflows. I think there were other, good, reasons we made this change too (originally the driver-specific scripts lived in driver-specific folders in this repository) but I cannot recall at the moment. @mbroadst do you recall?

Reply:
@ShaneHarvey I was worried that we would have a situation like we do with libmongocrypt here, where 12 different drivers were committing with different styles to a common repository. Rather, this project has inverted that design to expect certain "hooks" to be present in a given driver's repository. The hope was also that drivers would be able to reuse a lot of scripts they already have (e.g. node has an `install-dependencies.sh` script that already downloads a version of node and installs it globally), so the hook could just call that. 

At the end of the day I think the most complicated part is the case you bring up, but I'm not sure solving for that particular issue provides the most gain in terms of ease of implementation for all drivers. My expectation is that the variables you use in the axes here are things you already depend upon in your existing evergreen config.

Reply:
> where 12 different drivers were committing with different styles to a common repository. 

Hm, I see that as a non-issue especially because the driver tests are going to be released/published anywhere.

Another benefit of having it all live in this repo is that it becomes trivial to test different branches/versions of the driver in question (ie we could test pymongo 3.10, 3.11, master, etc...). With the current approach we would need to commit and maintain the test scripts in all branches we want to test.

Reply:
I am in the process of changing this so that driver-specific scripts live in this repo again. I agree with your rationale @ShaneHarvey - this is a much better pattern.

Reply:
Cool! Just one final note. Matt mentioned that he might prefer these tests to live in the node driver rather than this project for a couple reasons I won't list. But can't we have it both ways? PyMongo's tests can live in this repo and Node live in another repo.

Reply:
@ShaneHarvey we're just going to move everything over to this repo, I'd prefer to have a single solution for all drivers at this point rather than get fancy.